gpytorch/kernels/index_kernel.py

```diff
@@ -14,6 +14,8 @@ def _eval_covar_matrix(covar_factor, log_var):
 
 class IndexKernel(Kernel):
     def __init__(self, n_tasks, rank=1, prior=None, active_dims=None):
+        if rank > n_tasks:
+            raise RuntimeError("Cannot create a task covariance matrix larger than the number of tasks")
```

Comment:
I guess that's not exactly what it would be doing - it would just make the covar_factor fat, and so introduce unnecessary parameters - the matrix will still be of the correct size.

@rajkumarkarthik is pretty much done with an implementation of an LKJ prior (evaluate either over the full correlation matrix, or over its Cholesky factor). 

How do you feel about, in the case when `n_tasks = rank`, to parameterize the task kernel via its Cholesky factor? I guess the challenge then would be to enforce that during inference we stay in the subspace of Cholesky factors: `{L in R^{nxn} : L_ij = 0 for j>1, L_ii >0, and sum_j(Lij) = 1}`

Reply:
We parameterize our variational distributions by the Cholesky decomposition, so we could borrow code from that. Maybe we can open another issue for an IndexKernel refactor and push that to another PR since this one is pretty big already? I think the last issue to settle for this one is how the MultitaskMean should function (and then to actually use a ModuleList either way we go with it).

Reply:
Yeah sg I'll open up a new issue.

Also just commented on the MultitaskMean. I'll try to play around with this tomorrow a bit to make sure everything runs as expected for the things I've set up.

Reply:
No worries. It is the weekend after all -- I just had nothing better to do today than finish this up :-)